scrapy/spiders/sitemap.py

```diff
@@ -81,7 +81,7 @@ def _get_sitemap_body(self, response):
         # without actually being a .xml.gz file in the first place,
         # merely XML gzip-compressed on the fly,
         # in other word, here, we have plain XML
-        elif response.url.endswith('.xml') or response.url.endswith('.xml.gz'):
+        if re.search(r'\.xml(\.gz)?(\?|$)', response.url):
```

Comment:
Turning the `elif` into an `if` makes sense in this case, but then the previous `elif` should be changed as well, I think, to keep the method consistent.